controllers/vspherecluster_controller.go

```diff
@@ -72,15 +73,24 @@ func AddClusterControllerToManager(ctx context.Context, controllerManagerCtx *ca
 			},
 			NetworkProvider: networkProvider,
 		}
-		return ctrl.NewControllerManagedBy(mgr).
+		builder := ctrl.NewControllerManagedBy(mgr).
 			For(&vmwarev1.VSphereCluster{}).
 			WithOptions(options).
 			Watches(
 				&vmwarev1.VSphereMachine{},
 				handler.EnqueueRequestsFromMapFunc(reconciler.VSphereMachineToCluster),
 			).
-			WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), controllerManagerCtx.WatchFilterValue)).
-			Complete(reconciler)
+			WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), controllerManagerCtx.WatchFilterValue))
+
+		// Conditionally add a Watch for topologyv1.Zone when the feature gate is enabled
+		if feature.Gates.Enabled(feature.NamespaceScopedZones) {
+			builder = builder.Watches(
+				&topologyv1.Zone{},
+				handler.EnqueueRequestsFromMapFunc(reconciler.ZoneToVSphereClusters),
+			).WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), controllerManagerCtx.WatchFilterValue))
```

Comment:
```suggestion
			)
```
This one has been already added above

Reply:
Fixed！